tests/returntypes/test_autograd_qnode_new.py

```diff
@@ -0,0 +1,1942 @@
+        def circuit(a):
+            qml.RY(a[0], wires=0)
+            qml.RX(a[1], wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        a = np.array([0.1, 0.2], requires_grad=True)
+
+        grad = qml.grad(circuit)(a)
+
+        assert isinstance(grad, np.ndarray)
+        assert len(grad) == 2
+        assert grad.shape == (2,)
+
+    def test_execution_single_measurement_param_probs(self, dev_name, diff_method, mode):
+        """For a multi dimensional measurement (probs), check that a single array is returned with the correct
+        dimension"""
+        if diff_method == "adjoint":
+            pytest.skip("Test does not supports adjoint because of probabilities.")
+
+        dev = qml.device(dev_name, wires=2)
+
+        @qnode(dev, interface="autograd", diff_method=diff_method)
+        def circuit(a):
+            qml.RY(a, wires=0)
+            qml.RX(0.2, wires=0)
+            return qml.probs(wires=[0, 1])
+
+        a = np.array(0.1, requires_grad=True)
+
+        jac = qml.jacobian(circuit)(a)
+
+        assert isinstance(jac, np.ndarray)
+        assert jac.shape == (4,)
+
+    def test_execution_single_measurement_probs_multiple_param(self, dev_name, diff_method, mode):
+        """For a multi dimensional measurement (probs), check that a single tuple is returned containing arrays with
+        the correct dimension"""
+        if diff_method == "adjoint":
+            pytest.skip("Test does not supports adjoint because of probabilities.")
+
+        dev = qml.device(dev_name, wires=2)
+
+        @qnode(dev, interface="autograd", diff_method=diff_method)
+        def circuit(a, b):
+            qml.RY(a, wires=0)
+            qml.RX(b, wires=0)
+            return qml.probs(wires=[0, 1])
+
+        a = np.array(0.1, requires_grad=True)
+        b = np.array(0.2, requires_grad=True)
+
+        jac = qml.jacobian(circuit)(a, b)
+
+        assert isinstance(jac, tuple)
+
+        assert isinstance(jac[0], np.ndarray)
+        assert jac[0].shape == (4,)
+
+        assert isinstance(jac[1], np.ndarray)
+        assert jac[1].shape == (4,)
+
+    def test_execution_single_measurement_probs_multiple_param_single_array(
+        self, dev_name, diff_method, mode
+    ):
+        """For a multi dimensional measurement (probs), check that a single tuple is returned containing arrays with
+        the correct dimension"""
+        if diff_method == "adjoint":
+            pytest.skip("Test does not supports adjoint because of probabilities.")
+
+        dev = qml.device(dev_name, wires=2)
+
+        @qnode(dev, interface="autograd", diff_method=diff_method)
+        def circuit(a):
+            qml.RY(a[0], wires=0)
+            qml.RX(a[1], wires=0)
+            return qml.probs(wires=[0, 1])
+
+        a = np.array([0.1, 0.2], requires_grad=True)
+        jac = qml.jacobian(circuit)(a)
+
+        assert isinstance(jac, np.ndarray)
+        assert jac.shape == (4, 2)
```

Comment:
Return is not a tuple as mentioned in the docstring.

Reply:
I change the docstring